### DISCUSSION TITLE
Redirect to default IDP if user is not linked

### DISCUSSION THREAD

**Discussion by mm-pcosco** [sventorben/keycloak-home-idp-discovery#370] [Q&A] [answered]
Is it possible to always redirect to one default IdP if the entered username/email is not linked?
I only have one IDP configured in Keycloak. 
Users can't be uniquely identified through a domain, since any e-mail address is allowed in that IDP.

**Comment by sventorben** [comment]
Please check if this works for you --> https://github.com/sventorben/keycloak-home-idp-discovery/issues/34#issuecomment-1049176120

**Reply by mm-pcosco** [reply]
Thanks for your quick reply!

Unfortunately, this is not working on my side. I have federated and non-federated users in my Keycloak.

Whenever I type in a completely unknown username to Keylcoak it will just show the passwort field, instead of redirecting to a default IDP (on the IDP the user is already there).
![image](https://github.com/sventorben/keycloak-home-idp-discovery/assets/157100036/6001c257-e2f1-458b-b808-99317b28bce6)

However, if I configure a default identity provider in the `Identity Provider Redirector`, it will always redirect me to that default IDP, for any user (federated and non-federated)
![image](https://github.com/sventorben/keycloak-home-idp-discovery/assets/157100036/cabc0e60-3349-47a6-a863-b46177b7133d)

I'm using Keycloak v19 and [IDP-Discovery v.19.1.0](https://github.com/sventorben/keycloak-home-idp-discovery/releases/tag/v19.1.0).

**Reply by sventorben** [reply, marked as answer]
Ok, I see. You want to redirect non-existend users to an identity provider while existing local users should be able to enter their password, right?

That is currently not supported.

If this were supported by the extension, it would make it easy for attackers to discover registered emails present on keycloak, because the attacker would see different behaviour for non-existent and existent accounts.

**Reply by mm-pcosco** [reply]
Thanks for your clarification Sven.
